website/frontend/src/components/pages/Send/children/MessageForm/children/SwitchableField/SwitchableField.tsx

```diff
@@ -0,0 +1,37 @@
+import { ChangeEvent, useState } from 'react';
+import { useField } from 'formik';
+import { Input, Checkbox } from '@gear-js/ui';
+
+import styles from './SwitchableField.module.scss';
+
+type Props = {
+  name: string;
+  label: string;
+  defaultValue: string;
+};
+
+const SwitchableField = (props: Props) => {
+  const { name, label, defaultValue } = props;
+
+  const [field, , helpers] = useField(name);
+  const [isChecked, setIsChecked] = useState(false);
+
+  const toggleSwitch = (event: ChangeEvent<HTMLInputElement>) => {
+    const checked = event.target.checked;
+
+    if (!checked) {
+      helpers.setValue(defaultValue);
+    }
+
+    setIsChecked(checked);
+  };
```

Comment:
```suggestion
useEffect(() => {
  if (!isChecked) {
    helpers.setValue(initialValue);
  }
}, [isChecked])

  const toggle = () => {
    setIsChecked(prevValue => !prevValue);
  };
```
imo it's cleaner this way

Reply:
In my opinion it looks worse